addons/flannel/0.21.5/install.sh

```diff
@@ -307,20 +307,49 @@ function flannel_pods_remain() {
 }
 
 function remove_weave() {
-    # firstnode only
-    kubectl -n kube-system delete daemonset weave-net
-    kubectl -n kube-system delete rolebinding weave-net
-    kubectl -n kube-system delete role weave-net
-    kubectl delete clusterrolebinding weave-net
-    kubectl delete clusterrole weave-net
-    kubectl -n kube-system delete serviceaccount weave-net
-    kubectl -n kube-system delete secret weave-passwd
+    local resources=("daemonset.apps/weave-net"
+                     "rolebinding.rbac.authorization.k8s.io/weave-net"
+                     "role.rbac.authorization.k8s.io/weave-net"
+                     "clusterrolebinding.rbac.authorization.k8s.io/weave-net"
+                     "clusterrole.rbac.authorization.k8s.io/weave-net"
+                     "serviceaccount/weave-net"
+                     "secret/weave-passwd")
+
+    for resource in "${resources[@]}"; do
+        log "Waiting up to 5 minutes for weave resources to be deleted"
+        if ! spinner_until 320 kubectl -n kube-system delete "$resource" --ignore-not-found 1>/dev/null; then
```

Comment:
so this could take up to 7*320 = 37 minutes to finish for all resources? that is unacceptable.